### PR TITLE
Updated deprecated dependency request-promise-native

### PR DESCRIPTION
**Pull request by yashkohli88**
The service code has been updated to replace the deprecated dependency, request-promise-native, with axios. The following tasks have been performed:

- [x]  A fetch.js file was created in the lib folder. This file contains a function which takes request parameters as input, uses the axios library to initiate HTTP requests, and returns responses. It accepts method, URL, responseType ('json', 'text', 'stream'), headers, and data as request parameters.
- [x]  This function was integrated into the files of various folder wherever the request-promise-native library was previously invoked. Various files and functions were updated to accommodate this change.
- [x]  All replaced statements were individually tested, and an integration test suite was also used to ensure that no functionality was broken due to these changes.
- [x]  Test files were updated where stubs had been declared using request-promise-native.
- [x]  The request-promise-native library was removed from the package.json file.
- [x]  A test file for fetch.js, named fetchTests.js, was created to test the function with various scenarios. Please note that an internet connection is necessary to run these test cases.

Future scope -
- [ ] Refactor the code to declare the headers in centralised fetch.js file

This PR is related to the issue https://github.com/clearlydefined/service/issues/1090

### CODE REVIEW

File: business/statusService.js
Comment (yashkohli88) [inline business/statusService.js:1]: @qtomlinson I request your help to test all the requestPromise calls in this file.

Comment (qtomlinson) [reply business/statusService.js:1]: tested dev instance.

File: lib/pypiCoordinatesMapper.js
[the remark this answers is not in the document]
Comment (qtomlinson) [reply lib/pypiCoordinatesMapper.js:1]: tested against dev instance.  Also adding integration test for this.  See https://github.com/clearlydefined/operations/pull/89

File: providers/search/azureSearch.js
[the remark this answers is not in the document]
Comment (qtomlinson) [reply providers/search/azureSearch.js:1]: tested against dev instance. Also adding integration test for this. See https://github.com/clearlydefined/operations/pull/89

**Review by elrayle — APPROVED** [review]
Looks good.  Thanks for taking care of this technical debt.

**Pull request merged**